src/hermit.rs

```diff
@@ -43,6 +43,24 @@ impl<T: Config> Hermit<T> {
         let path = self.config.shell_root_path().join(name);
         file_ops.create_git_repo(path);
     }
+
+    pub fn inhabit_shell(&mut self, file_ops: &mut FileOperations, name: &str) -> Result<(), Error> {
+        //let mut shell_names = Vec::new();
+        //let mut res = Ok(());
+        //if name == "" {
+        //    shell_names = self.get_shell_list();
+        //} else {
+        let res = self.set_current_shell(name);
+        //}
+        match res {
+            Ok(v) => Ok(v),
+            Err(e) => Err(Error::ShellDoesNotExist),
+        }
```

Comment:
I think you can just return the result of `self.set_current_shell(name)`.  If it's an `Err` it should already contain `Error::ShellDoesNotExist` and you're just uwrapping and rewrapping the `Ok`.